services/dynamic-sidecar/src/simcore_service_dynamic_sidecar/api/containers.py

```diff
@@ -61,7 +63,7 @@ def _format_result(container_inspect: dict[str, Any]) -> dict[str, Any]:
 
         return container_inspect
 
-    async with docker_client() as docker:
+    async with container_inspect_lock, docker_client() as docker:
```

Comment:
Q: is this correct?  don't ``container_inpect_lock`` returns something as of:

```python
async with container_inspect_lock as lock, docker_client() as docker:
 ...
```

Besides, I personally think this is more clear (and I think it is equivalent, right?)
```python
async with container_inspect_lock:
    async with docker_client() as docker:
            ...
```

Reply:
So I've check and the `asyncio.Lock` will return `None` and is not intended to return any usable value (see extract below).

```python
class _ContextManagerMixin:
    async def __aenter__(self):
        await self.acquire()
        # We have no use for the "as ..."  clause in the with
        # statement for locks.
        return None

    async def __aexit__(self, exc_type, exc, tb):
        self.release()
```

We have mix of both styles in the code. I've refactored to this, because it reduces complexity by being less indented.